pkg/providers/tinkerbell/create.go

```diff
@@ -109,26 +109,39 @@ func (p *Provider) SetupAndValidateCreateCluster(ctx context.Context, clusterSpe
 		return err
 	}
 
+	if err := p.readCSVToCatalogue(); err != nil {
+		return err
+	}
+
 	spec := NewClusterSpec(clusterSpec, p.machineConfigs, p.datacenterConfig)
 
 	// TODO(chrisdoherty4) Look to inject the validator. Possibly look to use a builder for
 	// constructing the validations rather than injecting flags into the provider.
-	validator := NewClusterSpecValidator(
-		NewMinimumHardwareAvailableAssertion(p.catalogue),
+	clusterSpecValidator := NewClusterSpecValidator(
+		NewCreateMinimumHardwareAvailableAssertion(p.catalogue),
 	)
 
 	if !p.skipIpCheck {
-		validator.Register(NewIPNotInUseAssertion(p.netClient))
+		clusterSpecValidator.Register(NewIPNotInUseAssertion(p.netClient))
 	}
 
+	// Validate must happen last beacuse we depend on the catalogue entries for some checks.
+	if err := clusterSpecValidator.Validate(spec); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Provider) readCSVToCatalogue() error {
```

Comment:
This looks a little odd. To summarize, there's now the `readCSVToCatalogue()` function called on l.112 instead of having all the `readCSVToCatalogue()` logic in place.